src/Http/ApiProblem/ApiProblemFactory.php

```diff
@@ -116,11 +117,47 @@ public static function createFromThrowable(
             case $e instanceof EventCannotBeRemovedFromProduction:
                 return ApiProblem::blank($e->getMessage(), $e->getCode() ?: 400);
 
+            // Because almost any exception will be an instance of \Exception, we need to do a strict comparison of the
+            // class name here to convert generic \Exception exceptions specifically.
+            case get_class($e) === Exception::class:
+                return self::convertGenericExceptionToApiProblem($e, $request);
+
             default:
-                return ApiProblem::blank($e->getMessage(), $e->getCode() ?: 500);
+                return self::convertToGenericApiProblem($e);
         }
     }
 
+    private static function convertToGenericApiProblem(Throwable $e): ApiProblem
+    {
+        // Note: When we have improved the validation of most endpoints, we should make this return a plain
+        // "Internal Server Error" instead.
+        return ApiProblem::blank($e->getMessage(), $e->getCode() ?: 500);
+    }
+
+    /**
+     * Usually we will convert \Exception to a generic ApiProblem. However in some cases we want to convert generic
+     * exceptions thrown by external libraries to a more specific ApiProblem.
+     */
+    private static function convertGenericExceptionToApiProblem(
```

Comment:
Should the name of the method be  `convertExceptionToGenericApiProblem` to closer match `convertToGenericApiProblem` ?

Reply:
No because the "generic" is different in those two cases. In this case you convert a generic exception (`\Exception`) to an `ApiProblem`, which could be a specific one like the Bad Gateway or just a blank one for exceptions it doesn't know how to handle.

In `convertToGenericApiProblem` you convert the throwable to the blank ApiProblem for exceptions it doesn't kow how to handle.

But since I have to explain it the wording is not great so I'll see if I can improve it in some way :)

Reply:
I tried to improve it, let me know if this is clearer. Only thing I don't know how to improve/clarify is "generic exception" (to indicate `\Exception` specifically). Other things like "base exception", "literal exception", ... also feel vague.

Reply:
Perfect now, using `default` did the trick.